CMakeLists.txt

```diff
@@ -43,40 +43,6 @@ if (NOT EXISTS "${ClickHouse_SOURCE_DIR}/contrib/sysroot/README.md")
     message (FATAL_ERROR "Submodules are not initialized. Run\n\tgit submodule update --init")
 endif ()
 
-# Take care to add prlimit in command line before ccache, or else ccache thinks that
```

Comment:
We do this on purpose to avoid devs adding more and more complex files/templates that require a huge amount of time and resources to build. When we hit this it's a wake up call to look and fix those files, removing this means we can accumulate many bad changes and the build time can skyrocket

Reply:
thanks for the explanation. How do these issues manifest itself during the build? like this?:
```
2024-10-08T09:00:16.5396488Z Oct 08 11:00:16 prlimit --as=20000000000 --data=10000000000 --cpu=1000 /usr/bin/sccache /usr/bin/clang++-19 --target=x86_64-linux-gnu -- ... src/CMakeFiles/dbms.dir/Interpreters/Aggregator.cpp.o -MF src/CMakeFiles/dbms.dir/Interpreters/Aggregator.cpp.o.d -o src/CMakeFiles/dbms.dir/Interpreters/Aggregator.cpp.o -c /build/src/Interpreters/Aggregator.cpp
2024-10-08T09:00:16.5463778Z Oct 08 11:00:16 sccache: Compiler killed by signal 9
```

Reply:
Yes. Another example from the past: https://github.com/ClickHouse/ClickHouse/pull/63942#issuecomment-2120563112

Reply:
@thevar1able According to Raul these type of errors is subject to patch the code, not auto-restart ci pipeline. #70414.
I think CI just needs to return clear status that compilation takes more resources than allowed.

Reply:
Thank you. I'd expect this check to be stable, is it normal that it is flaky, @Algunenano?

Reply:
Happened again out of nowhere https://github.com/ClickHouse/ClickHouse/actions/runs/11276749452/job/31362795047?pr=70414